src/opm/parser/eclipse/EclipseState/Grid/MapAxes.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+  Copyright 2021 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <cmath>
+#include <string>
+
+#include <opm/common/utility/String.hpp>
+#include <opm/parser/eclipse/Units/Units.hpp>
+#include <opm/parser/eclipse/EclipseState/Grid/MapAxes.hpp>
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/io/eclipse/EclFile.hpp>
+#include <opm/io/eclipse/EclOutput.hpp>
+#include <opm/parser/eclipse/Parser/ParserKeywords/M.hpp>
+
+namespace Opm {
+
+namespace {
+
+double length_factor(const std::string& mapunits) {
+    auto mp = trim_copy(mapunits);
+    if (mp == "METRES")
+        return unit::meter;
+    else if (mp == "FEET")
+        return unit::feet;
+    else if (mp == "CM")
+        return prefix::centi*unit::meter;
+    throw std::logic_error("Unrecognized MAPUNITS value: " + mapunits);
```

Comment:
You should really include `<stdexcept>` to get a declaration of `std::logic_error` in scope.